src/workerd/server/workerd-api.c++

```diff
@@ -368,6 +368,13 @@ struct WorkerdApi::Impl final {
     if (maybeOwnedModuleRegistry != kj::none) {
       jsgIsolate.setUsingNewModuleRegistry();
     }
+
+    // Apply UTC timezone for Date methods
+    jsgIsolate.runInLockScope([&](JsgWorkerdIsolate::Lock& lock) {
+      lock.v8Isolate->DateTimeConfigurationChangeNotification(
+          v8::Isolate::TimeZoneDetection::kSkip);
+    });
```

Comment:
This could probably be set in `jsg/setup.h` when the isolate is created? Especially if we just want this to be the mode all isolates run in.